### ISSUE TITLE
"Bus error" on armhf Debian installed through Linux Deploy

### ISSUE THREAD

**Issue by imidazolium** [gansm/finalcut#1]
Thank you very much for this cool collection of widgets. They do look much more awesome than the ancient libnewt. I successfully compiled libfinalcut and the examples in test directory. Everything runs perfectly in x86_64 linux environment. 

However, on the armhf Debian on my tablet, the whole library as well as all examples compiled nicely but all the executable gave "**cqBus error**". I am wondering if there is anything in the source code restricted to x86_64/i386 archs.  

Maybe it is too greedy to ask for ARM support at the moment. :)

**Comment by gansm** [comment]
Thank you for your feedback. I have tested the Final Cut sources with a Debian Wheezy armhf image for QEMU form [https://people.debian.org/~aurel32/qemu/armhf/](https://people.debian.org/~aurel32/qemu/armhf/). This environment emulates a computer with a ARMv7 CPU. My compile tests showed that there is a typecast problem from the fc::SpecialCharacter enumeration to wchar_t. That bug has now been fixed.

I didn’t receive a cqBus error message during program execution. I think that this could be triggered by a SIGBUS signal from the kernel. This could indicate a memory failure or a problem with the MMU. Have you tried to start the program with the GNU Debugger or Valgrind?
